website/project/views/drafts.py

```diff
@@ -0,0 +1,210 @@
+# @must_be_valid_project
+# def submit_draft_for_review(auth, node, draft_pk, *args, **kwargs):
+#     user = auth.user
+#
+#     draft = get_draft_or_fail(draft_pk)
+#     draft.is_pending_review = True
+#     draft.save()
+#
+#     REVIEW_EMAIL = Mail(tpl_prefix='prereg_review', subject='New Prereg Prize registration ready for review')
+#     send_mail(draft.initiator.email, REVIEW_EMAIL, user=user, src=node)
+#
+#     ret = project_utils.serialize_node(node, auth)
+#     ret['success'] = True
+#     return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def draft_before_register_page(auth, node, draft, *args, **kwargs):
+    ret = serialize_node(node, auth, primary=True)
+
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+@http_error_if_disk_saving_mode
+def register_draft_registration(auth, node, draft, *args, **kwargs):
+
+    data = request.get_json()
+    register = draft.register(auth)
+
+    try:
+        if data.get('registrationChoice', 'immediate') == 'embargo':
+            # Initiate embargo
+            embargo_end_date = parse_date(data['embargoEndDate'], ignoretz=True)
+            register.embargo_registration(auth.user, embargo_end_date)
+        else:
+            register.require_approval(auth.user)
+        register.save()
+    except ValidationValueError as err:
+        raise HTTPError(http.BAD_REQUEST, data=dict(message_long=err.message))
+
+    push_status_message(language.AFTER_REGISTER_ARCHIVING,
+                        kind='info',
+                        trust=False)
+    return {
+        'status': 'initiated',
+        'urls': {
+            'registrations': node.web_url_for('node_registrations')
+        }
+    }, http.ACCEPTED
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registration(auth, node, draft, *args, **kwargs):
+    return serialize_draft_registration(draft, auth), http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registrations(auth, node, *args, **kwargs):
+
+    count = request.args.get('count', 100)
+    drafts = node.draft_registrations.find(
+        Q('registered_node', 'eq', None)
+    )[:count]
+    return {
+        'drafts': [serialize_draft_registration(d, auth) for d in drafts]
+    }, http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def create_draft_registration(auth, node, *args, **kwargs):
+
+    data = request.get_json()
+
+    schema_name = data.get('schema_name')
+    if not schema_name:
+        raise HTTPError(http.BAD_REQUEST)
+
+    schema_version = data.get('schema_version', 1)
+    schema_data = data.get('schema_data', {})
+
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', schema_version)
+    )
+    draft = node.create_draft_registration(auth.user, meta_schema, schema_data, save=True)
```

Comment:
If I am reading this correctly, no validation (against the intended schema) or sanitization (of submitted values) is performed. User-submitted data is being taken in directly without a further look.

That's a bit worrying on a few different levels- I'll take a look at the display layer, but be aware that our current standard is to run some layer of sanitization (`strip_html`) on entry before user inputs hit the database.